client/lib/checkout/tax-placeholders.js

```diff
@@ -0,0 +1,38 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get, has, isString, map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import debugFactory from 'debug';
+const debug = debugFactory( 'calypso:tax-placeholders' );
+
+export function maybeInjectTaxPlaceholdersIntoPurchase( purchase ) {
+	const purchasePriceText = get( purchase, 'price_text' );
+	if (
+		config.isEnabled( 'tax-placeholders' ) &&
+		! has( purchase, 'tax_text' ) &&
+		isString( purchasePriceText ) &&
+		! purchasePriceText.match( /include/i ) // "Included with plan"
+	) {
+		debug( 'injecting taxAmount into purchase', purchase );
+		return Object.assign( purchase, {
+			tax_amount: purchase.amount / 10,
+			tax_text: purchasePriceText.replace( /(?<!\d|\.)(\d+)(\d)(?:\.(\d*)\d)?/, '$1.$2$3' ),
+		} );
+	}
+	return purchase;
+}
+
+export function maybeInjectTaxPlaceholdersIntoPurchases( purchases ) {
```

Comment:
Took me a while to notice the `s` at the end of the function name. I thought there was some wild recursion going on!! 🖖

Reply:
Is that OK? Does it warrant a rename? (`maybeInjectTaxPlaceholdersIntoPurchaseArray`)?

Reply:
Or I could collapse into one function and branch on the type of the argument...

That would be nicer API, but probably doesn't matter for temporary code.

Reply:
> That would be nicer API, but probably doesn't matter for temporary code.

Understood. I'm reviewing _"unfiltered"_ today. Take everything with lots of salt grains.

Reply:
> I love it, don't stop :)
🕺